### DISCUSSION TITLE
What is the correct way to check if the current Thread "owns" a IScedulingRule?

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-platform/eclipse.platform#1483] [Q&A]
Assume i have the following

`ISchedulingRule rule = ResourcesPlugin.getWorkspace().getRoot();`

now I want to check if the current thread "owns" the rule already, similar to 

https://docs.oracle.com/javase/7/docs/api/java/lang/Thread.html#holdsLock%28java.lang.Object%29

I poked around a bit but didn't finde something in this regards, there is `Job.getJobManager().currentRule()` but from the description it will give different results depending on if its called from a Job and also depends on nesting rules...

**Comment by mickaelistria** [comment]
You can try something like
```java
try {
  jobManager.beginRule(rule, monitor);
  return true;
} catch (IllegalArgumentException ex) {
  return false;
} finally {
  manager.endRule(rule);
}
```

Or, with the currentRule, you can know if your operation can be scheduled or not in that thread calling by getting the currentRule and calling `!currentRule.isConflicting(myRule) || currentRule.contains(myRule)`
